### PR TITLE
Etag support

### PR DESCRIPTION
**Pull request by DarthSim**
Hey! Thanks for the project, it's very useful!
This PR adds Etags support. I'm not sure I patched everything I should yet this works for me, so maybe someone would find this useful.

### CODE REVIEW

File: fakestorage/upload.go [hunk 1]
```diff
@@ -127,14 +127,16 @@ func (s *Server) insertFormObject(r *http.Request) xmlResponse {
...
 	obj := Object{
 		ObjectAttrs: ObjectAttrs{
 			BucketName:      bucketName,
 			Name:            name,
 			ContentType:     contentType,
 			ContentEncoding: contentEncoding,
 			Crc32c:          checksum.EncodedCrc32cChecksum(data),
-			Md5Hash:         checksum.EncodedMd5Hash(data),
+			Md5Hash:         md5Hash,
+			Etag:            fmt.Sprintf("\"%s\"", md5Hash),
```
Comment (fsouza) [inline fakestorage/upload.go:139]: nit:

```suggestion
			Etag:            fmt.Sprintf("%q", md5Hash),
```

File: fakestorage/upload.go [hunk 2]
```diff
@@ -219,14 +221,16 @@ func (s *Server) simpleUpload(bucketName string, r *http.Request) jsonResponse {
...
 	obj := Object{
 		ObjectAttrs: ObjectAttrs{
 			BucketName:      bucketName,
 			Name:            name,
 			ContentType:     r.Header.Get(contentTypeHeader),
 			ContentEncoding: contentEncoding,
 			Crc32c:          checksum.EncodedCrc32cChecksum(data),
-			Md5Hash:         checksum.EncodedMd5Hash(data),
+			Md5Hash:         md5Hash,
+			Etag:            fmt.Sprintf("\"%s\"", md5Hash),
```
Comment (fsouza) [inline fakestorage/upload.go:233]:
```suggestion
			Etag:            fmt.Sprintf("%q", md5Hash),
```

File: fakestorage/upload.go [hunk 3]
```diff
@@ -261,14 +265,16 @@ func (s *Server) signedUpload(bucketName string, r *http.Request) jsonResponse {
...
```
Comment (fsouza) [inline fakestorage/upload.go:277]:
```suggestion
			Etag:            fmt.Sprintf("%q", md5Hash),
```

File: fakestorage/upload.go [hunk 4]
```diff
@@ -453,6 +461,7 @@ func (s *Server) uploadFileContent(r *http.Request) jsonResponse {
 	obj.Content = append(obj.Content, content...)
 	obj.Crc32c = checksum.EncodedCrc32cChecksum(obj.Content)
 	obj.Md5Hash = checksum.EncodedMd5Hash(obj.Content)
+	obj.Etag = fmt.Sprintf("\"%s\"", obj.Md5Hash)
```
Comment (fsouza) [inline fakestorage/upload.go:464]:
```suggestion
	obj.Etag = fmt.Sprintf("%q", obj.Md5Hash)
```

**Review by fsouza — COMMENTED** [review]
Thanks for contributing, just some nits on how we quote the string!

**Comment by DarthSim** [comment]
Yep, you're right, totally forgot about `%q`

**Pull request merged**